worker-workflow/src/main/java/com/github/cafdataprocessing/workflow/FailureFieldsManager.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2015-2018 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing.workflow;
+
+import com.google.gson.Gson;
+import com.hpe.caf.worker.document.model.Document;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class FailureFieldsManager
+{
+    private static final Logger LOG = LoggerFactory.getLogger(FailureFieldsManager.class);
+    private final Gson gson;
+
+    public FailureFieldsManager()
+    {
+        this.gson = new Gson();
+    }
+
+    public void handleExtraFailureSubFields(final Document document)
```

Comment:
Are we doing this for every document in the family?  Surely we really only need this set on the root document.  If so then I think taking in the `Task` rather than the `Document` would be clearer here.

Reply:
Workflow worker only ever acts on the root document for a family. It processes the entire family at once. Just like in the arguments manager, it takes in the `Document` object and added a field to it `CAF_WORKFLOW_SETTINGS`. We never loop over subdocuments inside the worker, I think it's fine being left at `Document` so that it is consistent.